7-writing-plans/README.md

```diff
@@ -4,162 +4,160 @@
+    
+    ```
+    2018-02-16T15:36:31.643418 DEBUG  Bolt::Inventory: Did not find node1 in inventory
+    2018-02-16T15:36:32.713360 DEBUG  Bolt::Executor: Started with 100 max thread(s)
+    2018-02-16T15:36:32.932771 DEBUG  Bolt::Inventory: Did not find node1 in inventory
+    2018-02-16T15:36:32.932869 INFO   Bolt::Executor: Starting task exercise7::write on ["node1"]
+    2018-02-16T15:36:32.932892 DEBUG  Bolt::Executor: Arguments: {"filename"=>"hello", "message"=>"world"} Input method: both
+    2018-02-16T15:36:33.178433 DEBUG  Bolt::Transport::SSH: Authentication method 'gssapi-with-mic' is not available
+    2018-02-16T15:36:33.179532 DEBUG  Bolt::Transport::SSH: Running task run 'Task({'name' => 'exercise7::write', 'executable' => '/Users/michaelsmith/puppetlabs/tasks-hands-on-lab/7-writing-plans/modules/exercise7/tasks/write.sh'})' on node1
+    Started on node1...
+    2018-02-16T15:36:33.216451 DEBUG  node1: Opened session
+    2018-02-16T15:36:33.216604 DEBUG  node1: Executing: mktemp -d
+    2018-02-16T15:36:33.395440 DEBUG  node1: stdout: /tmp/tmp.I7ZTz4OmfY
+    
+    2018-02-16T15:36:33.395746 DEBUG  node1: Command returned successfully
+    2018-02-16T15:36:33.411634 DEBUG  node1: Executing: chmod u+x '/tmp/tmp.I7ZTz4OmfY/write.sh'
+    2018-02-16T15:36:33.423831 DEBUG  node1: Command returned successfully
+    2018-02-16T15:36:33.424137 DEBUG  node1: Executing: PT_filename='hello' PT_message='world' '/tmp/tmp.I7ZTz4OmfY/write.sh'
+    2018-02-16T15:36:33.436180 DEBUG  node1: Command returned successfully
+    2018-02-16T15:36:33.436226 DEBUG  node1: Executing: rm -rf '/tmp/tmp.I7ZTz4OmfY'
+    2018-02-16T15:36:33.447658 DEBUG  node1: Command returned successfully
+    2018-02-16T15:36:33.447850 DEBUG  node1: Closed session
+    2018-02-16T15:36:33.447918 DEBUG  Bolt::Transport::SSH: Result on node1: {"_output":""}
+    Finished on node1:
+    
+      {
+      }
+    2018-02-16T15:36:33.448381 INFO   Bolt::Executor: Ran task 'exercise7::write' on 1 node with 0 failures
+    Ran on 1 node in 0.74 seconds
+    ```
+3. Write a plan that uses the task you created. Save the following as `modules/exercise7/plans/writeread.pp`:
+
+    ```puppet
+    plan exercise7::writeread (
+      TargetSpec $nodes,
+      String     $filename,
+      String     $message = 'Hello',
+    ) {
+      run_task(
+        'exercise7::write',
+        $nodes,
+        filename => $filename,
+        message  => $message,
+      )
+      run_command("cat /tmp/${filename}", $nodes)
+    }
+    ```
+
+    **Note:**
+    
+    * The plan takes three arguments, one of which (`message`) has a default value. We'll see shortly how Bolt uses that to validate user input.
+    * Use the Puppet `split` function to support passing a comma-separated list of nodes. Plans are just Puppet, so you can use any of the available [functions](https://docs.puppet.com/puppet/latest/function.html) or [native data types](https://docs.puppet.com/puppet/latest/lang_data_type.html).
```

Comment:
I think this comment is no longer relevant, since using the `TargetSpec` type makes comma-separated lists of nodes work already. This plan isn't using `split`.

Reply:
Removed.